experiments/download_ssl4eo.py

```diff
@@ -377,7 +374,7 @@ def get_patch_s2(
     )
 
 
-def get_random_patches_grid(
+def get_random_patches_rtree(
```

Comment:
How many times does this function get called? We want to build a single R-tree containing all `--indices_range` non-overlapping locations. We don't want to create a new R-tree every time this function gets called.

Reply:
Oh That's a bug here. The rtree object is passed as an argument, so every time using the same tree.

```
patches_s1, patches_s2c, patches_s2a, center_coord = get_random_patches_rtree(
            idx, collections, bands, crops, sampler, dates, radius, debug, rtree_obj
        )
```